### DISCUSSION TITLE
Template doesn't re-render

### DISCUSSION THREAD

**Discussion by nicnocquee** [vercel/next.js#64543] [Help]
### Summary

According to the docs, `Template` is like `Layout` but it's re-rendered on navigation. However after playing with it, I noticed that when navigating to a different path that uses the same `Page` component, it doesn't re-rendered. 

For example, two pages in `/app/admin/[id]/page.tsx` and `/app/guest/[id]/page.tsx` with a template `/app/template.tsx`. Say you're currently in `/admin/1` page. Navigating to `/guest/1` will re-render the template. However navigating to `/admin/2` from `/admin/1` doesn't re-render the template. Is this a bug or intentional?

A demo: https://www.nico.fyi/experiments/suspense-demo/with-template/admin/1 and accompanying blog post: https://www.nico.fyi/blog/understanding-layout-and-template-next-app-router

### Additional information

```js
`next info`:


Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:37 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6031
  Available memory (MB): 36864
  Available CPU cores: 14
Binaries:
  Node: 20.12.0
  npm: 10.5.0
  Yarn: N/A
  pnpm: 8.15.5
Relevant Packages:
  next: 14.2.1 // Latest available version is detected (14.2.1).
  eslint-config-next: 14.1.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.3
Next.js Config:
  output: N/A
```
```


### Example

_No response_

**Comment by rwieruch** [comment]
Related issue I think https://github.com/vercel/next.js/issues/60032
